src/transport/SecureSessionMgr.h

```diff
@@ -143,18 +149,17 @@ class DLL_EXPORT SecureSessionMgrBase : public Mdns::ResolveDelegate
      */
     System::Layer * SystemLayer() { return mSystemLayer; }
 
-protected:
     /**
      * @brief
      *   Initialize a Secure Session Manager
      *
      * @param localNodeId    Node id for the current node
      * @param systemLayer    System, layer to use
-     * @param transport Underlying Transport to use
      */
-    CHIP_ERROR InitInternal(NodeId localNodeId, System::Layer * systemLayer, Transport::Base * transport);
+    CHIP_ERROR Init(NodeId localNodeId, System::Layer * systemLayer);
 
 private:
+    friend class TransportMgrBase;
```

Comment:
We should avoid using friend as much as possible, I think we can use the delegate pattern to decouple the mutual reference between TransportMgrBase and SecureSessionMgr, just like what we did between SecureSessionMgr and ExchangeManager.

Only SecureSessionMgr martians a reference to TransportMgrBase, not the other way around, TransportMgrBase talks to SecureSessionMgr via delegate interface.

Reply:
+1. No friend classes at all would be the ideal goal. Worst case, we can have friends for testing purpuses or very focused inheritance.

In this case, it looks like we want to hide away the protected bits. I believe that encourages too much coupling.